sketch-on-map/src/main/java/com/esri/arcgismaps/sample/sketchonmap/MainActivity.kt

```diff
@@ -0,0 +1,299 @@
+        resetButtons()
+        activityMainBinding.pointLinePolygonToolbar.polygonButton.isSelected = true
+        geometryEditor?.apply {
+            tool = vertexTool
+            start(GeometryType.Polygon)
+        }
+    }
+
+    /**
+     * When the freehand line button is clicked, reset other buttons, show the freehand line
+     * button as selected, and start freehand line drawing mode.
+     */
+    fun createModeFreehandLine(view: View) {
+        setCurrentSelectionText("FreehandPolyline")
+        resetButtons()
+        activityMainBinding.pointLinePolygonToolbar.freehandLineButton.isSelected = true
+        geometryEditor?.apply {
+            tool = freehandTool
+            start(GeometryType.Polyline)
+        }
+    }
+
+    /**
+     * When the freehand polygon button is clicked, reset other buttons, show the freehand
+     * polygon button as selected, and enable freehand polygon drawing mode.
+     */
+    fun createModeFreehandPolygon(view: View) {
+        setCurrentSelectionText("FreehandPolygon")
+        resetButtons()
+        activityMainBinding.pointLinePolygonToolbar.freehandPolygonButton.isSelected = true
+        geometryEditor?.apply {
+            tool = freehandTool
+            start(GeometryType.Polygon)
+        }
+    }
+
+    /**
+     * When the undo button is clicked, undo the last event on the GeometryEditor.
+     */
+    fun undo(view: View) {
+        if (geometryEditor?.canUndo?.value == true) {
+            geometryEditor?.undo()
+            setCurrentSelectionText(getString(R.string.undo))
+        }
+    }
+
+    /**
+     * When the redo button is clicked, redo the last undone event on the GeometryEditor.
+     */
+    fun redo(view: View) {
+        if (geometryEditor?.canRedo?.value == true) {
+            geometryEditor?.redo()
+            setCurrentSelectionText(getString(R.string.redo))
+        }
+    }
+
+    /**
+     * When the stop button is clicked, check that sketch is valid. If so, get the geometry from
+     * the sketch, set its symbol and add it to the graphics overlay.
+     */
+    fun stop(view: View) {
+        // get the geometry from sketch editor
+        val sketchGeometry = geometryEditor?.geometry?.value
+            ?: return showError("Error retrieving geometry")
+
+        if (GeometryBuilder.builder(sketchGeometry)?.isSketchValid == false) {
+            return reportNotValid()
+        }
+
+        // stops the editing session
+        geometryEditor?.stop()
+
+        // clear button selection
+        resetButtons()
+
+        // create a graphic from the sketch editor geometry
+        val graphic = Graphic(sketchGeometry)
+        // assign a symbol based on geometry type
+        when (sketchGeometry) {
+            is Polygon -> graphic.symbol = fillSymbol
+            is Polyline -> graphic.symbol = lineSymbol
+            is Point, is Multipoint -> graphic.symbol = pointSymbol
+            else -> {}
+        }
```

Comment:
The setter could maybe be extracted from this `when` statement
```suggestion
        graphic.symbol = when (sketchGeometry) {
            is Polygon -> fillSymbol
            is Polyline -> lineSymbol
            is Point, is Multipoint -> pointSymbol
            else -> null
        }
```